### ISSUE TITLE
Possible issue with the nthRoot() function

### ISSUE THREAD

**Issue by sonnyk22** [josdejong/mathjs#490]
After passing the expressions below to the MathJS.eval() function, I get the exception shown below thrown back.

nthRoot(e ^ -2, e ^ -4):
![calc2](https://cloud.githubusercontent.com/assets/6744710/10738116/311e14b6-7bec-11e5-964a-1cc0f3c4d4d5.png)

nthRoot(-2, 3)
![calc3](https://cloud.githubusercontent.com/assets/6744710/10738120/33a3d6c6-7bec-11e5-8826-9e6427114ce3.png)

However, on other calculators such as Microsoft, I do get results back.
![calc1](https://cloud.githubusercontent.com/assets/6744710/10738156/5bb64b12-7bec-11e5-98b7-877f9f2db4c7.png)
![calc4](https://cloud.githubusercontent.com/assets/6744710/10738159/5d96e27a-7bec-11e5-8978-3d8478d50410.png)
![calc6](https://cloud.githubusercontent.com/assets/6744710/10738162/5f323af8-7bec-11e5-96a7-4158ac9ef8ad.png)

Is the expression that I am passing in to the MathJS.eval() function incorrect?

**Comment by josdejong** [comment]
Thanks for reporting this. I've done a little bit of debugging and found that JavaScripts `Math.pow` (used in an iteration in `nthroot`) doesn't always behave like we want. From [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/pow):
```js
// due to "even" and "odd" roots laying close to each other, 
// and limits in the floating number precision, 
// negative bases with fractional exponents always return NaN
Math.pow(-7, 1/3); // NaN
```

In this particular case the algorithm evaluates the following, resulting in NaN:
```js
Math.pow(-46.209093934213584, -0.9816843611112658); // NaN
```

So we have to see if we can improve `nthRoot` and work around this limitation in `Math.pow`.

**Comment by ericman314** [comment]
I will take a look at it tonight as well.

**Comment by josdejong** [comment]
Ok thanks @ericman314 , I don't have time tonight so that would be great...

**Comment by ericman314** [comment]
See #496.

**Comment by ericman314** [comment]
I submitted a pull request #496 which replaces the algorithm with this:

```js
// x = a ^ (1/root)
var x = Math.pow(Math.abs(a), 1/root);
// If a is negative and root is odd, then  (-1) ^ (1/root) = -1
x = a < 0 ? -x : x;
```

The square root algorithm (and cube root, etc.) is one of my favorite algorithms. The algorithm comes from Newton's method, which converges quickly but occasionally [runs into problems.](https://en.wikipedia.org/wiki/Newton%27s_method#Failure_analysis) `nthRoot(-2, 3)` just happens to result in a division by zero, although choosing an alternative starting point could fix that.

I thought that perhaps the algorithm may be faster than my suggestion above when root is an integer, since `pow` can be [optimized for integer exponents,](https://en.wikipedia.org/wiki/Exponentiation#Efficient_computation_with_integer_exponents) but the timing results below are not looking very good for the algorithm even with the (supposed) optimization:

```
80000 iterations of nthRoot(x, y), real-valued y
original: 6483ms
proposed: 15ms

80000 iterations of nthRoot(x, y), integer-valued y
original: 1186ms
original-optimized: 1257ms
proposed: 12ms

```

Even when using my own "optimized" version of `pow(real, integer)` (which is not actually any faster than the built-in implementation, no surprises there!) the algorithm is slower. It looks like the proposed solution is much faster in every case.

**Comment by josdejong** [comment]
This is fixed now in v2.4.1

**Comment by sonnyk22** [comment]
Thank you
